packages/react-scripts/config/webpack.config.prod.js

```diff
@@ -350,6 +368,43 @@ module.exports = {
     // https://github.com/jmblog/how-to-optimize-momentjs-with-webpack
     // You can remove this if you don't use Moment.js:
     new webpack.IgnorePlugin(/^\.\/locale$/, /moment$/),
+
+    // For some reason, Webpack adds some ids of all the modules that exist to our vendor chunk
+    // Instead of using numerical ids it uses a unique path to map our request to a module.
+    // Thanks to this change the vendor hash will now always stay the same
+    new webpack.NamedModulesPlugin(),
+
+    // Ensure that every chunks have an actual name and not an id
+    // If the chunk has a name, this name is used
+    // otherwise the name of the file is used
+    new webpack.NamedChunksPlugin(chunk => {
+      if (chunk.name) {
+        return chunk.name;
+      }
+      return chunk
+        .mapModules(m => m.request)
+        .map(filename => path.basename(filename, path.extname(filename)))
+        .join('_');
+    }),
+
+    // Avoid having the vendors in the rest of the app
+    // Only execute if the vendors file exists
+    ...activateWebpackPluginOnCondition(
```

Comment:
I'm a little hesitant with this approach as it's passed on to those who eject.

Sure, this is a really clever way of doing it (and my first thought, too), but it's a little indirect for ejecting purposes.

I feel like it wouldn't be as-bad if we inlined the condition, but I'd rather go even more idiomatic.

How about we make this:
```js
fs.existsSync(paths.appVendors) ? new webpack.optimize.CommonsChunkPlugin({
  name: 'vendors',
  minChunks: Infinity,
}) : null,
```

followed by adding a `.filter(Boolean)` to the array at the end (currently line 408).

That way people don't have to know what array spread is, and `filter(Boolean)` would weed out the null entry.
We can even note that it gets weeded out.